lib/tasks/create_mbrave_tags.rake

```diff
@@ -0,0 +1,215 @@
+    # mbrave.yml config needed by limber to be able to generate the mbrave UMI file at the
+    # end of the bioscan process.
+    class MbraveTagsCreator
+      attr_reader :forward_filename,
+                  :reverse_filename,
+                  :yaml_filename,
+                  :tag_identifier,
+                  :version,
+                  :forward_group,
+                  :reverse_groups
+
+      def initialize(params)
+        @forward_filename = params[:forward_filename]
+        @reverse_filename = params[:reverse_filename]
+        @tag_identifier = params[:tag_identifier]
+        @version = params[:version]
+        @yaml_filename = params[:yaml_filename]
+        @forward_group = nil
+        @reverse_groups = []
+      end
+
+      def reset_yaml
+        puts "Generating file #{yaml_filename}"
+        File.write(yaml_filename, {}.to_yaml)
+      end
+
+      def set_yaml_for_all_environments
+        contents = YAML.safe_load(File.read(yaml_filename))
+        new_contents = {}
+        new_contents['development'] = contents
+        new_contents['test'] = contents
+        new_contents['staging'] = contents
+        new_contents['production'] = contents
+        File.write(yaml_filename, new_contents.to_yaml)
+      end
+
+      def create_1_tag_group_forward
+        tags = []
+        mbrave_tags = []
+        puts "Creating forward_tags from: #{forward_filename}"
+        CSV.foreach(forward_filename, headers: true) do |row|
+          tag = Tag.new(map_id: row['Forward Index Number'], oligo: row['F index sequence'])
+          tags.push(tag)
+          mbrave_tags.push(row['Forward Oligo Label'])
+        end
+        puts " - #{forward_tag_group_name}"
+        @forward_group = _create_tag_group(forward_tag_group_name, tags)
+        _add_to_yaml(yaml_filename, forward_tag_group_name, mbrave_tags, version)
+      end
+
+      def create_24_tag_groups_reverse
+        tags = []
+        mbrave_tags = []
+        group = 1
+        puts "Creating reverse tags from: #{reverse_filename}"
+        CSV.foreach(reverse_filename, headers: true) do |row|
+          map_id = row['Reverse Index Number'].to_i
+          pos = ((map_id - 1) % 4)
+          tag = Tag.new(map_id: pos, oligo: row['R index sequence'])
+          tags.push(tag)
+          mbrave_tags.push(row['Reverse Oligo Label'])
+
+          if pos == 3
+            puts " - #{reverse_tag_group_name(group)}"
+            @reverse_groups.push(_create_tag_group(reverse_tag_group_name(group), tags))
+            _add_to_yaml(yaml_filename, reverse_tag_group_name(group), mbrave_tags, version)
+            group += 1
+            tags = []
+            mbrave_tags = []
+          end
+        end
+      end
+
+      def create_tag_layout_templates
+        puts 'Creating tag layout templates:'
+        @reverse_groups.each_with_index do |reverse_group, index|
+          puts " - #{tag_layout_template_name(index)}"
+          TagLayoutTemplate.create(
```

Comment:
What happens if the tag group or tag plate layout has already been generated before for this version number? e.g. if there's a mistake and we try to create a v3 set when a v3 set already exists.

Reply:
It will break on tag group creation.